src/test/java/org/broadinstitute/hellbender/utils/smithwaterman/SWPairwiseAlignmentUnitTest.java

```diff
@@ -117,4 +117,119 @@ public void  testForIdenticalAlignmentsWithDifferingFlankLengths() {
+        final String read      = "AAAAAAA";
+        final int expectedStart = 0;
+        final String expectedCigar = "1M358D6M29D";
+        final SWPairwiseAlignment sw = new SWPairwiseAlignment(reference.getBytes(), read.getBytes(),
+                                                               SWPairwiseAlignment.ORIGINAL_DEFAULT,
+                                                               SWPairwiseAlignment.OverhangStrategy.INDEL);
+        sw.printAlignment(reference.getBytes(), read.getBytes());
+        Assert.assertEquals(sw.getAlignmentStart2wrt1(), expectedStart);
+        Assert.assertEquals(sw.getCigar().toString(), expectedCigar);
+    }
+
+    @Test(enabled = true)
+    public void testSubstringMatchLeadingIndelLong() {
+        final String reference = "ATAGAAAATAGTTTTTGGAAATATGGGTGAAGAGACATCTCCTCTTATGGAAAAAGGGATTCTAGAATTTAACAATAAATATTCCCAACTTTCCCCAAGGCTTTAAAATCTACCTTGAAGGAGCAGCTGATGTATTTCTAGAACAGACTTAGGTGTCTTGGTGTGGCCTGTAAAGAGATACTGTCTTTCTCTTTTGAGTGTAAGAGAGAAAGGACAGTCTACTCAATAAAGAGTGCTGGGAAAACTGAATATCCACACACAGAATAATAAAACTAGATCCTATCTCTCACCATATACAAAGATCAACTCAAAACAAATTAAAGACCTAAATGTAAGACAAGAAATTATAAAACTACTAGAAAAAAACACAAGGGAAATGCTTCAGGACATTGGC";
+        final String read      = "AAAAAAA";
+        final int expectedStart = 0;
+        final String expectedCigar = "1M1D6M";
+        final SWPairwiseAlignment sw = new SWPairwiseAlignment(reference.getBytes(), read.getBytes(),
+                                                               SWPairwiseAlignment.ORIGINAL_DEFAULT,
+                                                               SWPairwiseAlignment.OverhangStrategy.LEADING_INDEL);
+        sw.printAlignment(reference.getBytes(), read.getBytes());
+        Assert.assertEquals(sw.getAlignmentStart2wrt1(), expectedStart);
+        Assert.assertEquals(sw.getCigar().toString(), expectedCigar);
+    }
+
+    @Test(enabled = true)
+    public void testSubstringMatchLeadingIgnoreLong() {
+        final String reference = "ATAGAAAATAGTTTTTGGAAATATGGGTGAAGAGACATCTCCTCTTATGGAAAAAGGGATTCTAGAATTTAACAATAAATATTCCCAACTTTCCCCAAGGCTTTAAAATCTACCTTGAAGGAGCAGCTGATGTATTTCTAGAACAGACTTAGGTGTCTTGGTGTGGCCTGTAAAGAGATACTGTCTTTCTCTTTTGAGTGTAAGAGAGAAAGGACAGTCTACTCAATAAAGAGTGCTGGGAAAACTGAATATCCACACACAGAATAATAAAACTAGATCCTATCTCTCACCATATACAAAGATCAACTCAAAACAAATTAAAGACCTAAATGTAAGACAAGAAATTATAAAACTACTAGAAAAAAACACAAGGGAAATGCTTCAGGACATTGGC";
+        final String read      = "AAAAAAA";
+        final int expectedStart = 359;
+        final String expectedCigar = "7M";
+        final SWPairwiseAlignment sw = new SWPairwiseAlignment(reference.getBytes(), read.getBytes(),
```

Comment:
\[FindBugs\] WARNING: Dm: Found reliance on default encoding in org\.broadinstitute\.hellbender\.utils\.smithwaterman\.SWPairwiseAlignmentUnitTest\.testSubstringMatchLeadingIgnoreLong\(\): String\.getBytes\(\)